### ISSUE TITLE
clusterctl create cluster - apply objects only to internal cluster

### ISSUE THREAD

**Issue by jessicaochen** [kubernetes-sigs/cluster-api#343]
There are some objects that should only be applied to the external cluster (eg. storage class)

Currently these are applied in a manual kubectl apply step after clusterctl is done running. https://github.com/kubernetes-sigs/cluster-api/blob/264cdcf9789eae783adc14a0008b07d71a3d18d1/clusterctl/examples/google/addons.yaml.template

It would be great if there was a flag in the clusterctl create command to apply these so that when the clusterctl command is finished, the cluster is set up.

It is up for debate whether this should be a yaml that gets applied or something like a patch that modifies provider-components.yaml.
------
TLDR - I currently favor Yaml
[Yaml]
Pro - Straightforward syntax. Makes sure that provider-components.yaml is consistent between the external and internal cluster.
[Patch]
Pro - Allows customizing provider-components.yaml  setting for the internal cluster. However, it is not good to diverge the external and internal stack too much as we cannot expect consistent behavior with very different settings.

**Comment by roberthbailey** [comment]
This was fixed in https://github.com/kubernetes-sigs/cluster-api/pull/376.
